### DISCUSSION TITLE
How to Download HTML file to share

### DISCUSSION THREAD

**Discussion by KMac0615** [executablebooks/meta#162] [Q&A]
Hello. I am trying to use jupyter book to generate a single HTML file to share with my team. I am working in JupyterHub and can build the file just fine using ```jb build my_file.ipynb```. When I click on the generated HMTL file, it opens in my browser and everything is rendered fine. However, I am trying to download the HTML file from JupyterHub to share. When I download and open the file in a browser, nothing is rendered correctly. None of my graphs show up, I just see "_images/my_file.png". I see paragraph symbols after my section headers. It is not showing up at all like it does when I open it from JupyterHub.

Side note, I cannot share this report through GitHub, or any other version control, as it contains sensitive information so this is not allowed by my company. I am looking for a better way of generating a report from an ipynb than having to save my graphs as images and embed them into a markdown cell in my notebook. Jupyter book seemed like a good answer to this issue so any help on being able to download and share the generated HTML files is greatly appreciated! Thank you.

**Comment by choldgraf** [comment]
ah - that's because it's not generating a *single* HTML file, it is generating a *collection* of files (AKA a website). Generating a single HTML file is more difficult because you'd have to embed all of the assets needed to make it up (CSS files, JS files, images, etc), and right now this isn't supported. So to share a notebook as a single file, your best bet is to use a PDF output *via* HTML, or to just share the notebook directly. I think it'd be great to support single-HTML building but I'm not quite sure how to accomplish it...

**Reply by choldgraf** [reply]
maybe @chrisjsewell or @mmcky has heard of a Sphinx workflow that embeds all of the assets in a single page?

**Reply by chrisjsewell** [reply]
not that I know of.
The current way, I guess, would be to share the full build folder

**Reply by mmcky** [reply]
there is a `singlehtml` builder in `sphinx` https://www.sphinx-doc.org/en/master/man/sphinx-build.html#options but it only groups the `html` into a single page and still requires the support of `css`, `js`, `image` files etc.

**Reply by akhmerov** [reply]
Perhaps it's possible to combine a singlehtml builder with other tools that inline HTML resources. A quick search finds a couple of browser extensions and some npm packages, like [this one](https://github.com/jonathantneal/posthtml-inline-assets).

**Reply by chrisjsewell** [reply]
yep good point thanks

**Reply by KMac0615** [reply]
Thanks all for the quick response! I appreciate the guidance.

**Reply by choldgraf** [reply]
ah it would be fantastic if we could get a pattern working like what @mmcky and @akhmerov describe. A lot of people would benefit from having a single HTML file to share with others

**Reply by choldgraf** [reply]
I opened up https://github.com/executablebooks/jupyter-book/issues/1046 to track this enhancement
